contracts/pool/AbstractPool.sol

```diff
@@ -106,30 +110,12 @@ abstract contract AbstractPool is PoolFDT, Pausable, Ownable {
 
     /// @notice withdraws the caller's liquidity assets
     /// @param  _amount the amount of LA to be withdrawn
-    function withdraw(uint256 _amount) public whenNotPaused returns (bool) {
-        require(balanceOf(msg.sender) >= _amount, "P:INSUFFICIENT_BALANCE");
-        require(depositDate[msg.sender] + poolInfo.lockupPeriod <= block.timestamp, "P:FUNDS_LOCKED");
-
-        // Check if the current withdrawal exceeds the limit in the specified period
-        if (block.timestamp < lastWithdrawalTime[msg.sender] + withdrawPeriod) {
-            require(lastWithdrawalAmount[msg.sender] + _amount <= withdrawLimit, "P:WITHDRAW_LIMIT_EXCEEDED");
-        } else {
-            lastWithdrawalAmount[msg.sender] = 0;
-        }
-
-        // Update the withdrawal history
-        lastWithdrawalTime[msg.sender] = block.timestamp;
-        lastWithdrawalAmount[msg.sender] += _amount;
-
-        /**
-         * TODO: Tigran Arakelyan - possible attack vector
-         * Investor can withdraw big amount partially and contract will be happy with it
-         * Maybe we should check cumulative withdrawals amount
-         */
-        if (_amount > poolInfo.withdrawThreshold) {
-            emit WithdrawalOverThreshold(msg.sender, _amount);
-            revert("P:THRESHOLD_REACHED");
-        }
+    /// @param  _index the index of the DepositInstance
+    function withdraw(uint256 _amount, uint256 _index) public whenNotPaused returns (bool) {
```

Comment:
1. It would be better UX if we could provide an array of indexes, so we can withdraw all unlocked funds just in one transaction (gas consumption wise, UX wise) 

2. Also we see uint256 as overkill for index value. We suggest using at least uint16 (see bellow)

`function withdraw(uint256 _amount, uint16[] _indexes) public whenNotPaused returns (bool) {`